tests/qos/test_pfc_pause.py

```diff
@@ -0,0 +1,348 @@
+    @param max_test_intfs_count: maximum count of interfaces to test. By default, it is a very large value to cover all the interfaces.  
+    return: Return # of iterations and # of passed iterations for each tested interface.   
+    """
+    dut_hostname = testbed['dut']
+    dut_ans = AnsibleHost(ansible_adhoc, dut_hostname)
+    
+    ptf_hostname = testbed['ptf']
+    ptf_ans = AnsibleHost(ansible_adhoc, ptf_hostname)
+    
+    """ Clear DUT's PFC counters """
+    dut_ans.sonic_pfc_counters(method="clear")
+    
+    """ Disable DUT's PFC wd """
+    dut_ans.shell('sudo pfcwd stop')
+    
+    """ Configure T0 tesbted and return testbed information """
+    dut_intfs, ptf_intfs, ptf_ip_addrs, ptf_mac_addrs = config_testbed_t0(ansible_adhoc, testbed)
+    
+    results = dict()
+
+    for i in range(min(max_test_intfs_count, len(ptf_intfs))):
+        src_index = i
+        dst_index = (i + 1) % len(ptf_intfs)
+        
+        src_intf = ptf_intfs[src_index]
+        dst_intf = ptf_intfs[dst_index]
+        
+        src_ip = ptf_ip_addrs[src_index]
+        dst_ip = ptf_ip_addrs[dst_index]
+        
+        src_mac = ptf_mac_addrs[src_index]
+        dst_mac = ptf_mac_addrs[dst_index]
+       
+        """ DUT interface to pause """
+        dut_intf_paused = dut_intfs[dst_index]
+                                
+        """ Clear MAC table in DUT (host memory and ASIC) """
+        dut_ans.shell('sonic-clear fdb all </dev/null >/dev/null 2>&1 &')
+        dut_ans.shell('sudo ip -s -s neigh flush all </dev/null >/dev/null 2>&1 &')
+        time.sleep(2)
+        
+        """ Populate the MAC table """
+        dut_ans.shell('ping -c 2 %s </dev/null >/dev/null 2>&1 &' % (src_ip))
+        dut_ans.shell('ping -c 2 %s </dev/null >/dev/null 2>&1 &' % (dst_ip))
+        time.sleep(2)
+        
+        """ Ensure the MAC table is correct """
+        if not check_mac_table(dut_ans, [src_mac, dst_mac]):
+            print 'MAC table of DUT is incorrect'
+            continue 
+        
+        if send_pause:            
+            peer_device = conn_graph_facts['device_conn'][dut_intf_paused]['peerdevice']
+            peer_port = conn_graph_facts['device_conn'][dut_intf_paused]['peerport']
+            peer_port_name = eos_to_linux_intf(peer_port)
+            peerdev_ans = AnsibleHost(ansible_adhoc, peer_device)
+        
+            cmd = "nohup sudo python %s -i %s -g -t %d -n %d </dev/null >/dev/null 2>&1 &" % (PFC_GEN_FILE_DEST, peer_port_name, pause_time, PFC_PKT_COUNT)
+        
+            if pfc_pause:
+                cmd = "nohup sudo python %s -i %s -p %d -t %d -n %d </dev/null >/dev/null 2>&1 &" % (PFC_GEN_FILE_DEST, peer_port_name, 2**pause_prio, pause_time, PFC_PKT_COUNT)
+                
+            """ Start PFC / FC storm """
+            peerdev_ans.shell(cmd)
+       
+            """ Wait for PFC pause frame generation """
+            time.sleep(2)
```

Comment:
These should be in a helper function. you are going to need it in multiple tests

Reply:
Fixed.